[Ngooi Jun Sen] iP

junngithub opened the pull request:
# **Icarus:** Your faithful companion

Have _you_ ever found yourself completing a task way too close to a deadline and wondered, "Wow, I wished I had something to remind me". Fret not, Icarus is here to help. Knowing the dangers of flying too close to the Sun, Icarus is here to help you with:
- Managing tasks
- Managing deadlines (coming soon)
- Reminders (coming soon)

# Why Icarus?
Icarus is:
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

Best of all, he is here whenever you need it.

# How to get started?
Download the latest release from [here](https://github.com/junngithub/ip/releases).


🌞  Don't fly too close to the Sun! Give Icarus a try today! 🚀

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

junngithub committed:
level 0

junngithub committed:
level 1 commit

junngithub committed:
Level 2 commit

junngithub committed:
level 3 commit

junngithub committed:
made task abstract, added child classes

junngithub committed:
Automated Text UI Testing

junngithub committed:
added exceptions

junngithub committed:
rectified todo, added delete function

junngithub committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

junngithub committed:
Add save feature

Feature saves the tasks onto "./data/icarus.txt", loads data from "./data/icarus.txt".

junngithub committed:
Make the code more OOP and divide classes into packages

junngithub committed:
Fix updateSave function in Storage bug and create factory constructors for Tasks.

updateSave did not save Tasks in the right format. Shifted input checking to respective Task factory constructors as createTask in Parser is getting too long

junngithub committed:
Update .gitignore to ignore .txt file that Tasks are saved onto

lauhengyi commented on src/main/java/Icarus.java:
```diff
@@ -0,0 +1,255 @@
+        readCommand();
+    }
+
+    private void returnNumberOfItems() {
+        System.out.println("You have " + list.size() + " item(s) in your list.");
+    }
```
Is it better if the naming of methods stay more consistent? The previous method (sayBye) uses the word say to connotate that the method writes to command line. Perhaps doing so for other corresponding functions might make the code more readable?

lauhengyi reviewed (commented):
Overall, code quality looks decent. Your naming is mostly descriptive and understandable.

junngithub committed:
Teach the chatbot how to understand dates and times.

junngithub committed:
Tweak the code to comply with a coding standard and clean up code to remove unnecessary imports

junngithub committed:
Give users a way to find a task by searching for a keyword in the task description

junngithub committed:
Add JavaDoc comments to the code

junngithub committed:
Merge branch 'branch-Level-9'

junngithub committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/Commands/AddCommand.java
#	src/main/java/Icarus.java

junngithub committed:
Clean up code after merge

junngithub committed:
Add JUnit tests to test the behavior of the code

blacksesamezongzi commented on src/main/java/Exceptions/EmptyInputException.java:
```diff
@@ -0,0 +1,42 @@
+        case "number" -> this.problem = " requires a number.";
+        case "missing by" -> this.problem = " task requires '/by'.";
+        case "missing from" -> this.problem = " task requires '/from'.";
+        case "missing to" -> this.problem = " task requires '/to'.";
```
I like the way you set the case line as un-intended, good job!

blacksesamezongzi commented on src/main/java/Exceptions/InvalidInputException.java:
```diff
@@ -0,0 +1,53 @@
+     *
+     * @param index the index of the item that was attempted to be accessed.
+     * @param size the total number of items in the task list.
+     */
```
The explanation is clear and precise.

blacksesamezongzi commented on src/main/java/Tasks/Events.java:
```diff
@@ -0,0 +1,92 @@
+     * "/from" or "/to" keywords, or start or end dates.
+     * @throws InvalidInputException if the input start or end dates' format is invalid.
+     */
+    public static Task of(String string, Parser parser)
```
Perhaps can add some blank lines to show multiple blocks?

blacksesamezongzi commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,79 @@
+     * @return a string representing the task in a file-compatible format.
+     */
+    public String toFile() {
+        int i;
```
Should the variable only be declared when the value is assigned?

blacksesamezongzi reviewed (commented):
LGTM. Just a few nits to fix.

junngithub committed:
Add a GUI to the chatbot and refactor functions that were too long

junngithub committed:
Use checkStyle to detect coding style violations

junngithub committed:
Use assertions to document important assumptions that should hold at various points in the code

junngithub committed:
Improve code quality

junngithub committed:
Merge pull request #3 from junngithub/branch-A-Assertions

Use assertions to document important assumptions that should hold at …

junngithub committed:
Merge pull request #2 from junngithub/branch-A-CodeQuality

Improve code quality

junngithub committed:
Make the command syntax more flexible

Writing out the whole command may be tedious. It is good to give users
the ability to shorten frequently used commands e.g. deadline, delete,
as it improves user experience.

junngithub committed:
Improve the GUI to make it more polished

Previous GUI was barebones. Tweaking the GUI to make it more pleasing
to look at enhances user experience. Users will be also able to resize
the window and ensure the content resize appropriately as the window
changes size.

junngithub committed:
Fix a bug related to opening tasks.txt and add user guide

Previous commit did not consider how parsing tasks from tasks.txt
file and tasks from user input require separate handling. User
guide added to let users understand how to use the Icarus chatbot.

junngithub committed:
Update MainWindow so it closes after 1 second

Even after user types "bye" which triggers ExitCommand, the window
still remains open, which may be confusing to the user.

junngithub committed:
Update README

junngithub committed:
Update JavaDoc
